core/src/main/java/org/elasticsearch/ingest/PipelineStore.java

```
@@ -154,18 +90,64 @@ public void close() throws IOException {
+                pipelines.put(pipeline.getId(), constructPipeline(pipeline.getId(), pipeline.getConfigAsMap()));
+            } catch (Exception e) {
+                throw new RuntimeException(e);
+            }
+        }
+        this.pipelines = Collections.unmodifiableMap(pipelines);
+    }
+
     /**
      * Deletes the pipeline specified by id in the request.
      */
-    public void delete(DeletePipelineRequest request, ActionListener<DeleteResponse> listener) {
-        ensureReady();
+    public void delete(DeletePipelineRequest request, ActionListener<WritePipelineResponse> listener) {
+        clusterService.submitStateUpdateTask("delete-pipeline-" + request.id(), new AckedClusterStateUpdateTask<WritePipelineResponse>(request, listener) {
 
-        DeleteRequest deleteRequest = new DeleteRequest(request);
-        deleteRequest.index(PipelineStore.INDEX);
-        deleteRequest.type(PipelineStore.TYPE);
-        deleteRequest.id(request.id());
-        deleteRequest.refresh(true);
-        client.delete(deleteRequest, handleWriteResponseAndReloadPipelines(listener));
+            @Override
+            protected WritePipelineResponse newResponse(boolean acknowledged) {
+                return new WritePipelineResponse(acknowledged);
+            }
+
+            @Override
+            public ClusterState execute(ClusterState currentState) throws Exception {
+                return innerDelete(request, currentState);
+            }
+        });
+    }
+
+    ClusterState innerDelete(DeletePipelineRequest request, ClusterState currentState) {
+        IngestMetadata currentIngestMetadata = currentState.metaData().custom(IngestMetadata.TYPE);
+        if (currentIngestMetadata == null) {
+            return currentState;
+        }
+        Map<String, PipelineConfiguration> pipelines = currentIngestMetadata.getPipelines();
+        if (pipelines.containsKey(request.id()) == false) {
+            // nothing to delete...
+            return currentState;
```

shall we instead return an error / 404 when the pipeline is not there? I think that would be appropriate for a CRUD api?